.github/workflows/ci.yml

```diff
@@ -417,21 +417,21 @@ jobs:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@v1
         with:
-          toolchain: stable
-          components: clippy
+          toolchain: nightly
```

Comment:
We were specifically running `clippy` from the `stable` release channel so that our lints don't just start failing overnight for no reason. Lints in the `nightly` version of `clippy` are also not guaranteed to be stabilized, so I don't see much point adhering to those rules.

Reply:
I don't think the crates will check with stable. Previously, clippy ran with the host machine as target, so std was available, but now (with `cd`-ing into the folder instead of `--manifest-path` `cargo` will pick up the `.cargo/config.toml`, set the correct target and there stable will fail as `build-std` is not stable.

Reply:
I guess we could run clippy with the `+esp` toolchain so that it can pick up the unstable stuff while only checking the stable lints?